src/arizona_socket.erl

```diff
@@ -52,7 +46,7 @@ Components state.
 
 -spec new(View, Assigns) -> t()
     when View :: view(),
-         Assigns :: assigns().
+         Assigns :: arizona_template_renderer:assigns().
```

Comment:
So there's a real difference between "macros" and "assigns"? (but they just happen to be represented by the same internal type, at the moment?)

Reply:
"Macros" substitutes/expands "assigns" at the compile time of the template, e.g.:
```erlang
-module(compile_example).
-export([compile/1, render/1]).

compile(Macros) ->
    arizona_template_compiler:compile(?MODULE, render, Macros).

render(Macros) ->
    maybe
        {ok, Tokens} ?= arizona_template_scanner:scan(~"""
        <div>{_@foo}</div>
        """),
        {ok, Elems} ?= arizona_template_parser:parse(Tokens),
        {ok, {Elems, Macros}}
    else
        {error, Reason} ->
            {error, Reason}
    end.
```

`$ rebar3 shell`:
```erlang
1> compile_example:compile(#{}). % <-- empty macros
{ok,#{function => render,
      id => [0],
      module => compile_example,
      static => [<<"<div>">>,<<"</div>">>], % <-- foo will always be evaluated and must be defined in "assigns"
      changeable =>
          #{0 =>
                {expr,#{function => #Fun<erl_eval.42.39164016>,
                        id => [0,0],
                        vars => [foo]}}},
      changeable_indexes => [0],
      changeable_vars => #{foo => [[0]]}, % <-- foo is "a var that changes"
      norm_assigns => #{},norm_assigns_vars => []}}
2> compile_example:compile(#{foo => foo}). % <-- foo defined as a macro
{ok,#{function => render,
      id => [0],
      module => compile_example,
      static => [<<"<div>foo</div>">>], % <-- foo was expanded
      changeable => #{},changeable_indexes => [],
      changeable_vars => #{},norm_assigns => #{}, % <-- no variables
      norm_assigns_vars => []}}
```

"Macros" and "assigns" are the same thing but different.

Reply:
> the same thing but different.

I've always liked stuff like this 😆